DIPs/DIPxxxx_attributes.md

```diff
@@ -0,0 +1,185 @@
+```
+pure: nothrow: @nogc: @safe:
+```
+at the top of a file means that one can never "undo" those attributes.
+This DIP proposes a solution to easily change the default values of the attributes while also allowing symbol specific 
+overrides of individual attribute groups.
+
+## Description
+
+Move all (DMD) compiler recognized attributes into `core.attribute`, making them symbols in their own right, 
+grouping by attribute groups into `enum`s, each with
+* a value `inferred`. The compiler shall determine the value of the attribute. It is illegal to have a function declaration `inferred`.
+* the attribute(s)
+* (if applicable) the attributes' logical negation.
+
+It is suggested that 
+* `inferred` have the value 0, and the compiler shall be required to replace enum core attributes
+that have the value `inferred` to the acual value as determined by the compiler, i.e. no symbols in object files or in reflection shall
+have a core attribute with the value `inferred`. 
+* the current default have the value 1.
+* the negation of the default (`@safe` for `@safe`/`@system`/`@trusted`) the value 2.
+* any other values (e.g. `@trusted`) continue from the value 3.
+
+A module declaration may be tagged with zero or more attribute groups, to apply to all symbols (bar templates which remain inferred with explicit tagging) declared within the module acting as the default.
+If any attribute groups are absent, then the value for that attribute group default to the corresponding value in `core.attribute.defaultAttributeSet`, which will have the values of the current defauls, but may be versioned in druntime as the end user wishes, or with command line switches (e.g. `-safe` or if Type_Info / Module Info generation is added as an attribute `-betterC`).
+
+As all the attributes are now symbols we can group the in an `AliasSeq` like fashion to apply them en masse, as is done in LDC for [`@fastmath`](https://github.com/ldc-developers/druntime/blob/ldc/src/ldc/attributes.d#L58).
+
+It is illegal to explicitly provide more than one attribute from any given attribute group as they are mutually exclusive. 
+Attributes applied explicity to any symbol override the module default attribute set.
+
+### Attributes & attribute like compiler behaviour encompassed in this DIP
+
+Encompassed:
+
+* pure
+* @nothrow
+* @nogc
+* @safe/@system/@trusted
+
+Optionally encompassed:
+
+* final
+* Type_Info / Module Info generation (other components of -betterC?)
+
+Not encompassed:
+
+* @disable
+* @property
+
+### Breaking changes / deprecation process
+
+Use of the current attributes that are not prefiex by an `@` such as `pure` and `nothrow`,
+and optionally other modifiers that are attribute like such as `final` will be changed to refer to the `core.attribute` symbols,
+and thus their use without the leading `@` will be deprecated. 
+
+No breaking changes are intended, although the introduction of the new enum symbols to be implicitly imported by `object.d`
```

Comment:
Yes I was getting myself confused between unintended breaking changes and scheduled deprecating changes.